File: Components/WinForms/CS/XpoTutorials/ODataV3Service/ODataClient.cs
namespace DevExpress.Xpo.Demos.OData {
    public partial class NorthwindEntities : global::System.Data.Services.Client.DataServiceContext {
        public NorthwindEntities(global::System.Uri serviceRoot) :
            base(serviceRoot) {
            this.ResolveName = new global::System.Func<global::System.Type, string>(this.ResolveNameFromType);
            this.ResolveType = new global::System.Func<string, global::System.Type>(this.ResolveTypeFromName);
            this.OnContextCreated();
        }
        partial void OnContextCreated();
        protected global::System.Type ResolveTypeFromName(string typeName) {
            if(typeName.StartsWith("NorthwindModel", global::System.StringComparison.Ordinal)) {
                return this.GetType().Assembly.GetType(string.Concat("DevExpress.Xpo.Demos", typeName.Substring(14)), false);
            }
            return null;
        }
        protected string ResolveNameFromType(global::System.Type clientType) {
            if(clientType.Namespace.Equals("DevExpress.Xpo.Demos", global::System.StringComparison.Ordinal)) {
                return string.Concat("NorthwindModel.", clientType.Name);
            }
            return null;
        }
        public global::System.Data.Services.Client.DataServiceQuery<Customers> Customers {
            get {
                if((this._Customers == null)) {
                    this._Customers = base.CreateQuery<Customers>("Customers");
                }
                return this._Customers;
            }
        }
        private global::System.Data.Services.Client.DataServiceQuery<Customers> _Customers;
        public global::System.Data.Services.Client.DataServiceQuery<Orders> Orders {
            get {
                if((this._Orders == null)) {
                    this._Orders = base.CreateQuery<Orders>("Orders");
                }
                return this._Orders;
            }
        }
        private global::System.Data.Services.Client.DataServiceQuery<Orders> _Orders;
        public void AddToCustomers(Customers customers) {
            base.AddObject("Customers", customers);
        }
        public void AddToOrders(Orders orders) {
            base.AddObject("Orders", orders);
        }
    }
    [global::System.Data.Services.Common.EntitySetAttribute("Customers")]
    [global::System.Data.Services.Common.DataServiceKeyAttribute("CustomerID")]
    public partial class Customers : global::System.ComponentModel.INotifyPropertyChanged {
        public static Customers CreateCustomers(string customerID, string companyName) {
            Customers customers = new Customers();
            customers.CustomerID = customerID;
            customers.CompanyName = companyName;
            return customers;
        }
        public string CustomerID {
            get {
                return this._CustomerID;
            }
            set {
                this.OnCustomerIDChanging(value);
                this._CustomerID = value;
                this.OnCustomerIDChanged();
                this.OnPropertyChanged("CustomerID");
            }
        }
        private string _CustomerID;
        partial void OnCustomerIDChanging(string value);
        partial void OnCustomerIDChanged();
        public string CompanyName {
            get {
                return this._CompanyName;
            }
            set {
                this.OnCompanyNameChanging(value);
                this._CompanyName = value;
                this.OnCompanyNameChanged();
                this.OnPropertyChanged("CompanyName");
            }
        }
        private string _CompanyName;
        partial void OnCompanyNameChanging(string value);
        partial void OnCompanyNameChanged();
        public string ContactName {
            get {
                return this._ContactName;
            }
            set {
                this.OnContactNameChanging(value);
                this._ContactName = value;
                this.OnContactNameChanged();
                this.OnPropertyChanged("ContactName");
            }
        }
        private string _ContactName;
        partial void OnContactNameChanging(string value);
        partial void OnContactNameChanged();
        public string ContactTitle {
            get {
                return this._ContactTitle;
            }
            set {
                this.OnContactTitleChanging(value);
                this._ContactTitle = value;
                this.OnContactTitleChanged();
                this.OnPropertyChanged("ContactTitle");
            }
        }
        private string _ContactTitle;
        partial void OnContactTitleChanging(string value);
        partial void OnContactTitleChanged();
        public string Address {
            get {
                return this._Address;
            }
            set {
                this.OnAddressChanging(value);
                this._Address = value;
                this.OnAddressChanged();
                this.OnPropertyChanged("Address");
            }
        }
        private string _Address;
        partial void OnAddressChanging(string value);
        partial void OnAddressChanged();
        public string City {
            get {
                return this._City;
            }
            set {
                this.OnCityChanging(value);
                this._City = value;
                this.OnCityChanged();
                this.OnPropertyChanged("City");
            }
        }
        private string _City;
        partial void OnCityChanging(string value);
        partial void OnCityChanged();
        public string Region {
            get {
                return this._Region;
            }
            set {
                this.OnRegionChanging(value);
                this._Region = value;
                this.OnRegionChanged();
                this.OnPropertyChanged("Region");
            }
        }
        private string _Region;
        partial void OnRegionChanging(string value);
        partial void OnRegionChanged();
        public string PostalCode {
            get {
                return this._PostalCode;
            }
            set {
                this.OnPostalCodeChanging(value);
                this._PostalCode = value;
                this.OnPostalCodeChanged();
                this.OnPropertyChanged("PostalCode");
            }
        }
        private string _PostalCode;
        partial void OnPostalCodeChanging(string value);
        partial void OnPostalCodeChanged();
        public string Country {
            get {
                return this._Country;
            }
            set {
                this.OnCountryChanging(value);
                this._Country = value;
                this.OnCountryChanged();
                this.OnPropertyChanged("Country");
            }
        }
        private string _Country;
        partial void OnCountryChanging(string value);
        partial void OnCountryChanged();
        public string Phone {
            get {
                return this._Phone;
            }
            set {
                this.OnPhoneChanging(value);
                this._Phone = value;
                this.OnPhoneChanged();
                this.OnPropertyChanged("Phone");
            }
        }
        private string _Phone;
        partial void OnPhoneChanging(string value);
        partial void OnPhoneChanged();
        public string Fax {
            get {
                return this._Fax;
            }
            set {
                this.OnFaxChanging(value);
                this._Fax = value;
                this.OnFaxChanged();
                this.OnPropertyChanged("Fax");
            }
        }
        private string _Fax;
        partial void OnFaxChanging(string value);
        partial void OnFaxChanged();
        public event global::System.ComponentModel.PropertyChangedEventHandler PropertyChanged;
        protected virtual void OnPropertyChanged(string property) {
            if((this.PropertyChanged != null)) {
                this.PropertyChanged(this, new global::System.ComponentModel.PropertyChangedEventArgs(property));
            }
        }
    }
    [global::System.Data.Services.Common.EntitySetAttribute("Orders")]
    [global::System.Data.Services.Common.DataServiceKeyAttribute("OrderID")]
    public partial class Orders : global::System.ComponentModel.INotifyPropertyChanged {
        public static Orders CreateOrders(int orderID) {
            Orders orders = new Orders();
            orders.OrderID = orderID;
            return orders;
        }
        public int OrderID {
            get {
                return this._OrderID;
            }
            set {
                this.OnOrderIDChanging(value);
                this._OrderID = value;
                this.OnOrderIDChanged();
                this.OnPropertyChanged("OrderID");
            }
        }
        private int _OrderID;
        partial void OnOrderIDChanging(int value);
        partial void OnOrderIDChanged();
        public global::System.Nullable<global::System.DateTime> OrderDate {
            get {
                return this._OrderDate;
            }
            set {
                this.OnOrderDateChanging(value);
                this._OrderDate = value;
                this.OnOrderDateChanged();
                this.OnPropertyChanged("OrderDate");
            }
        }
        private global::System.Nullable<global::System.DateTime> _OrderDate;
        partial void OnOrderDateChanging(global::System.Nullable<global::System.DateTime> value);
        partial void OnOrderDateChanged();
        public global::System.Nullable<global::System.DateTime> RequiredDate {
            get {
                return this._RequiredDate;
            }
            set {
                this.OnRequiredDateChanging(value);
                this._RequiredDate = value;
                this.OnRequiredDateChanged();
                this.OnPropertyChanged("RequiredDate");
            }
        }
        private global::System.Nullable<global::System.DateTime> _RequiredDate;
        partial void OnRequiredDateChanging(global::System.Nullable<global::System.DateTime> value);
        partial void OnRequiredDateChanged();
        public global::System.Nullable<global::System.DateTime> ShippedDate {
            get {
                return this._ShippedDate;
            }
            set {
                this.OnShippedDateChanging(value);
                this._ShippedDate = value;
                this.OnShippedDateChanged();
                this.OnPropertyChanged("ShippedDate");
            }
        }
        private global::System.Nullable<global::System.DateTime> _ShippedDate;
        partial void OnShippedDateChanging(global::System.Nullable<global::System.DateTime> value);
        partial void OnShippedDateChanged();
        public global::System.Nullable<int> ShipVia {
            get {
                return this._ShipVia;
            }
            set {
                this.OnShipViaChanging(value);
                this._ShipVia = value;
                this.OnShipViaChanged();
                this.OnPropertyChanged("ShipVia");
            }
        }
        private global::System.Nullable<int> _ShipVia;
        partial void OnShipViaChanging(global::System.Nullable<int> value);
        partial void OnShipViaChanged();
        public global::System.Nullable<decimal> Freight {
            get {
                return this._Freight;
            }
            set {
                this.OnFreightChanging(value);
                this._Freight = value;
                this.OnFreightChanged();
                this.OnPropertyChanged("Freight");
            }
        }
        private global::System.Nullable<decimal> _Freight;
        partial void OnFreightChanging(global::System.Nullable<decimal> value);
        partial void OnFreightChanged();
        public string ShipName {
            get {
                return this._ShipName;
            }
            set {
                this.OnShipNameChanging(value);
                this._ShipName = value;
                this.OnShipNameChanged();
                this.OnPropertyChanged("ShipName");
            }
        }
        private string _ShipName;
        partial void OnShipNameChanging(string value);
        partial void OnShipNameChanged();
        public string ShipAddress {
            get {
                return this._ShipAddress;
            }
            set {
                this.OnShipAddressChanging(value);
                this._ShipAddress = value;
                this.OnShipAddressChanged();
                this.OnPropertyChanged("ShipAddress");
            }
        }
        private string _ShipAddress;
        partial void OnShipAddressChanging(string value);
        partial void OnShipAddressChanged();
        public string ShipCity {
            get {
                return this._ShipCity;
            }
            set {
                this.OnShipCityChanging(value);
                this._ShipCity = value;
                this.OnShipCityChanged();
                this.OnPropertyChanged("ShipCity");
            }
        }
        private string _ShipCity;
        partial void OnShipCityChanging(string value);
        partial void OnShipCityChanged();
        public string ShipRegion {
            get {
                return this._ShipRegion;
            }
            set {
                this.OnShipRegionChanging(value);
                this._ShipRegion = value;
                this.OnShipRegionChanged();
                this.OnPropertyChanged("ShipRegion");
            }
        }
        private string _ShipRegion;
        partial void OnShipRegionChanging(string value);
        partial void OnShipRegionChanged();
        public string ShipPostalCode {
            get {
                return this._ShipPostalCode;
            }
            set {
                this.OnShipPostalCodeChanging(value);
                this._ShipPostalCode = value;
                this.OnShipPostalCodeChanged();
                this.OnPropertyChanged("ShipPostalCode");
            }
        }
        private string _ShipPostalCode;
        partial void OnShipPostalCodeChanging(string value);
        partial void OnShipPostalCodeChanged();
        public string ShipCountry {
            get {
                return this._ShipCountry;
            }
            set {
                this.OnShipCountryChanging(value);
                this._ShipCountry = value;
                this.OnShipCountryChanged();
                this.OnPropertyChanged("ShipCountry");
            }
        }
        private string _ShipCountry;
        partial void OnShipCountryChanging(string value);
        partial void OnShipCountryChanged();
        public event global::System.ComponentModel.PropertyChangedEventHandler PropertyChanged;
        protected virtual void OnPropertyChanged(string property) {
            if((this.PropertyChanged != null)) {
                this.PropertyChanged(this, new global::System.ComponentModel.PropertyChangedEventArgs(property));
            }
        }
    }
}

File: Components/WinForms/VB/XpoTutorials/ODataV3Service/ODataClient.vb
Imports Microsoft.VisualBasic
Imports System
Namespace DevExpress.Xpo.Demos.ODataV3
    Partial Public Class NorthwindEntities
        Inherits System.Data.Services.Client.DataServiceContext
        Public Sub New(ByVal serviceRoot As Global.System.Uri)
            MyBase.New(serviceRoot)
            Me.ResolveName = New Global.System.Func(Of Global.System.Type, String)(AddressOf Me.ResolveNameFromType)
            Me.ResolveType = New Global.System.Func(Of String, Global.System.Type)(AddressOf Me.ResolveTypeFromName)
            Me.OnContextCreated()
        End Sub
        Partial Private Sub OnContextCreated()
        End Sub
        Protected Function ResolveTypeFromName(ByVal typeName As String) As Global.System.Type
            If typeName.StartsWith("NorthwindModel", Global.System.StringComparison.Ordinal) Then
                Return Me.GetType().Assembly.GetType(String.Concat("DevExpress.Xpo.Demos", typeName.Substring(14)), False)
            End If
            Return Nothing
        End Function
        Protected Function ResolveNameFromType(ByVal clientType As Global.System.Type) As String
            If clientType.Namespace.Equals("DevExpress.Xpo.Demos", Global.System.StringComparison.Ordinal) Then
                Return String.Concat("NorthwindModel.", clientType.Name)
            End If
            Return Nothing
        End Function
        Public ReadOnly Property Customers() As Global.System.Data.Services.Client.DataServiceQuery(Of Customers)
            Get
                If (Me._Customers Is Nothing) Then
                    Me._Customers = MyBase.CreateQuery(Of Customers)("Customers")
                End If
                Return Me._Customers
            End Get
        End Property
        Private _Customers As Global.System.Data.Services.Client.DataServiceQuery(Of Customers)
        Public ReadOnly Property Orders() As Global.System.Data.Services.Client.DataServiceQuery(Of Orders)
            Get
                If (Me._Orders Is Nothing) Then
                    Me._Orders = MyBase.CreateQuery(Of Orders)("Orders")
                End If
                Return Me._Orders
            End Get
        End Property
        Private _Orders As Global.System.Data.Services.Client.DataServiceQuery(Of Orders)
        Public Sub AddToCustomers(ByVal customers As Customers)
            MyBase.AddObject("Customers", customers)
        End Sub
        Public Sub AddToOrders(ByVal orders As Orders)
            MyBase.AddObject("Orders", orders)
        End Sub
    End Class
    <Global.System.Data.Services.Common.EntitySetAttribute("Customers"), Global.System.Data.Services.Common.DataServiceKeyAttribute("CustomerID")> _
    Partial Public Class Customers
        Implements System.ComponentModel.INotifyPropertyChanged
        Public Shared Function CreateCustomers(ByVal customerID As String, ByVal companyName As String) As Customers
            Dim customers As New Customers()
            customers.CustomerID = customerID
            customers.CompanyName = companyName
            Return customers
        End Function
        Public Property CustomerID() As String
            Get
                Return Me._CustomerID
            End Get
            Set(ByVal value As String)
                Me.OnCustomerIDChanging(value)
                Me._CustomerID = value
                Me.OnCustomerIDChanged()
                Me.OnPropertyChanged("CustomerID")
            End Set
        End Property
        Private _CustomerID As String
        Partial Private Sub OnCustomerIDChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnCustomerIDChanged()
        End Sub
        Public Property CompanyName() As String
            Get
                Return Me._CompanyName
            End Get
            Set(ByVal value As String)
                Me.OnCompanyNameChanging(value)
                Me._CompanyName = value
                Me.OnCompanyNameChanged()
                Me.OnPropertyChanged("CompanyName")
            End Set
        End Property
        Private _CompanyName As String
        Partial Private Sub OnCompanyNameChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnCompanyNameChanged()
        End Sub
        Public Property ContactName() As String
            Get
                Return Me._ContactName
            End Get
            Set(ByVal value As String)
                Me.OnContactNameChanging(value)
                Me._ContactName = value
                Me.OnContactNameChanged()
                Me.OnPropertyChanged("ContactName")
            End Set
        End Property
        Private _ContactName As String
        Partial Private Sub OnContactNameChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnContactNameChanged()
        End Sub
        Public Property ContactTitle() As String
            Get
                Return Me._ContactTitle
            End Get
            Set(ByVal value As String)
                Me.OnContactTitleChanging(value)
                Me._ContactTitle = value
                Me.OnContactTitleChanged()
                Me.OnPropertyChanged("ContactTitle")
            End Set
        End Property
        Private _ContactTitle As String
        Partial Private Sub OnContactTitleChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnContactTitleChanged()
        End Sub
        Public Property Address() As String
            Get
                Return Me._Address
            End Get
            Set(ByVal value As String)
                Me.OnAddressChanging(value)
                Me._Address = value
                Me.OnAddressChanged()
                Me.OnPropertyChanged("Address")
            End Set
        End Property
        Private _Address As String
        Partial Private Sub OnAddressChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnAddressChanged()
        End Sub
        Public Property City() As String
            Get
                Return Me._City
            End Get
            Set(ByVal value As String)
                Me.OnCityChanging(value)
                Me._City = value
                Me.OnCityChanged()
                Me.OnPropertyChanged("City")
            End Set
        End Property
        Private _City As String
        Partial Private Sub OnCityChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnCityChanged()
        End Sub
        Public Property Region() As String
            Get
                Return Me._Region
            End Get
            Set(ByVal value As String)
                Me.OnRegionChanging(value)
                Me._Region = value
                Me.OnRegionChanged()
                Me.OnPropertyChanged("Region")
            End Set
        End Property
        Private _Region As String
        Partial Private Sub OnRegionChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnRegionChanged()
        End Sub
        Public Property PostalCode() As String
            Get
                Return Me._PostalCode
            End Get
            Set(ByVal value As String)
                Me.OnPostalCodeChanging(value)
                Me._PostalCode = value
                Me.OnPostalCodeChanged()
                Me.OnPropertyChanged("PostalCode")
            End Set
        End Property
        Private _PostalCode As String
        Partial Private Sub OnPostalCodeChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnPostalCodeChanged()
        End Sub
        Public Property Country() As String
            Get
                Return Me._Country
            End Get
            Set(ByVal value As String)
                Me.OnCountryChanging(value)
                Me._Country = value
                Me.OnCountryChanged()
                Me.OnPropertyChanged("Country")
            End Set
        End Property
        Private _Country As String
        Partial Private Sub OnCountryChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnCountryChanged()
        End Sub
        Public Property Phone() As String
            Get
                Return Me._Phone
            End Get
            Set(ByVal value As String)
                Me.OnPhoneChanging(value)
                Me._Phone = value
                Me.OnPhoneChanged()
                Me.OnPropertyChanged("Phone")
            End Set
        End Property
        Private _Phone As String
        Partial Private Sub OnPhoneChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnPhoneChanged()
        End Sub
        Public Property Fax() As String
            Get
                Return Me._Fax
            End Get
            Set(ByVal value As String)
                Me.OnFaxChanging(value)
                Me._Fax = value
                Me.OnFaxChanged()
                Me.OnPropertyChanged("Fax")
            End Set
        End Property
        Private _Fax As String
        Partial Private Sub OnFaxChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnFaxChanged()
        End Sub
        Public Event PropertyChanged As Global.System.ComponentModel.PropertyChangedEventHandler Implements System.ComponentModel.INotifyPropertyChanged.PropertyChanged
        Protected Overridable Sub OnPropertyChanged(ByVal [property] As String)
            If (Me.PropertyChangedEvent IsNot Nothing) Then
                RaiseEvent PropertyChanged(Me, New Global.System.ComponentModel.PropertyChangedEventArgs([property]))
            End If
        End Sub
    End Class
    <Global.System.Data.Services.Common.EntitySetAttribute("Orders"), Global.System.Data.Services.Common.DataServiceKeyAttribute("OrderID")> _
    Partial Public Class Orders
        Implements System.ComponentModel.INotifyPropertyChanged
        Public Shared Function CreateOrders(ByVal orderID As Integer) As Orders
            Dim orders As New Orders()
            orders.OrderID = orderID
            Return orders
        End Function
        Public Property OrderID() As Integer
            Get
                Return Me._OrderID
            End Get
            Set(ByVal value As Integer)
                Me.OnOrderIDChanging(value)
                Me._OrderID = value
                Me.OnOrderIDChanged()
                Me.OnPropertyChanged("OrderID")
            End Set
        End Property
        Private _OrderID As Integer
        Partial Private Sub OnOrderIDChanging(ByVal value As Integer)
        End Sub
        Partial Private Sub OnOrderIDChanged()
        End Sub
        Public Property OrderDate() As Global.System.DateTime?
            Get
                Return Me._OrderDate
            End Get
            Set(ByVal value? As System.DateTime)
                Me.OnOrderDateChanging(value)
                Me._OrderDate = value
                Me.OnOrderDateChanged()
                Me.OnPropertyChanged("OrderDate")
            End Set
        End Property
        Private _OrderDate? As Global.System.DateTime
        Partial Private Sub OnOrderDateChanging(ByVal value? As Global.System.DateTime)
        End Sub
        Partial Private Sub OnOrderDateChanged()
        End Sub
        Public Property RequiredDate() As Global.System.DateTime?
            Get
                Return Me._RequiredDate
            End Get
            Set(ByVal value? As System.DateTime)
                Me.OnRequiredDateChanging(value)
                Me._RequiredDate = value
                Me.OnRequiredDateChanged()
                Me.OnPropertyChanged("RequiredDate")
            End Set
        End Property
        Private _RequiredDate? As Global.System.DateTime
        Partial Private Sub OnRequiredDateChanging(ByVal value? As Global.System.DateTime)
        End Sub
        Partial Private Sub OnRequiredDateChanged()
        End Sub
        Public Property ShippedDate() As Global.System.DateTime?
            Get
                Return Me._ShippedDate
            End Get
            Set(ByVal value? As System.DateTime)
                Me.OnShippedDateChanging(value)
                Me._ShippedDate = value
                Me.OnShippedDateChanged()
                Me.OnPropertyChanged("ShippedDate")
            End Set
        End Property
        Private _ShippedDate? As Global.System.DateTime
        Partial Private Sub OnShippedDateChanging(ByVal value? As Global.System.DateTime)
        End Sub
        Partial Private Sub OnShippedDateChanged()
        End Sub
        Public Property ShipVia() As Integer?
            Get
                Return Me._ShipVia
            End Get
            Set(ByVal value? As Integer)
                Me.OnShipViaChanging(value)
                Me._ShipVia = value
                Me.OnShipViaChanged()
                Me.OnPropertyChanged("ShipVia")
            End Set
        End Property
        Private _ShipVia? As Integer
        Partial Private Sub OnShipViaChanging(ByVal value? As Integer)
        End Sub
        Partial Private Sub OnShipViaChanged()
        End Sub
        Public Property Freight() As Decimal?
            Get
                Return Me._Freight
            End Get
            Set(ByVal value? As Decimal)
                Me.OnFreightChanging(value)
                Me._Freight = value
                Me.OnFreightChanged()
                Me.OnPropertyChanged("Freight")
            End Set
        End Property
        Private _Freight? As Decimal
        Partial Private Sub OnFreightChanging(ByVal value? As Decimal)
        End Sub
        Partial Private Sub OnFreightChanged()
        End Sub
        Public Property ShipName() As String
            Get
                Return Me._ShipName
            End Get
            Set(ByVal value As String)
                Me.OnShipNameChanging(value)
                Me._ShipName = value
                Me.OnShipNameChanged()
                Me.OnPropertyChanged("ShipName")
            End Set
        End Property
        Private _ShipName As String
        Partial Private Sub OnShipNameChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnShipNameChanged()
        End Sub
        Public Property ShipAddress() As String
            Get
                Return Me._ShipAddress
            End Get
            Set(ByVal value As String)
                Me.OnShipAddressChanging(value)
                Me._ShipAddress = value
                Me.OnShipAddressChanged()
                Me.OnPropertyChanged("ShipAddress")
            End Set
        End Property
        Private _ShipAddress As String
        Partial Private Sub OnShipAddressChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnShipAddressChanged()
        End Sub
        Public Property ShipCity() As String
            Get
                Return Me._ShipCity
            End Get
            Set(ByVal value As String)
                Me.OnShipCityChanging(value)
                Me._ShipCity = value
                Me.OnShipCityChanged()
                Me.OnPropertyChanged("ShipCity")
            End Set
        End Property
        Private _ShipCity As String
        Partial Private Sub OnShipCityChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnShipCityChanged()
        End Sub
        Public Property ShipRegion() As String
            Get
                Return Me._ShipRegion
            End Get
            Set(ByVal value As String)
                Me.OnShipRegionChanging(value)
                Me._ShipRegion = value
                Me.OnShipRegionChanged()
                Me.OnPropertyChanged("ShipRegion")
            End Set
        End Property
        Private _ShipRegion As String
        Partial Private Sub OnShipRegionChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnShipRegionChanged()
        End Sub
        Public Property ShipPostalCode() As String
            Get
                Return Me._ShipPostalCode
            End Get
            Set(ByVal value As String)
                Me.OnShipPostalCodeChanging(value)
                Me._ShipPostalCode = value
                Me.OnShipPostalCodeChanged()
                Me.OnPropertyChanged("ShipPostalCode")
            End Set
        End Property
        Private _ShipPostalCode As String
        Partial Private Sub OnShipPostalCodeChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnShipPostalCodeChanged()
        End Sub
        Public Property ShipCountry() As String
            Get
                Return Me._ShipCountry
            End Get
            Set(ByVal value As String)
                Me.OnShipCountryChanging(value)
                Me._ShipCountry = value
                Me.OnShipCountryChanged()
                Me.OnPropertyChanged("ShipCountry")
            End Set
        End Property
        Private _ShipCountry As String
        Partial Private Sub OnShipCountryChanging(ByVal value As String)
        End Sub
        Partial Private Sub OnShipCountryChanged()
        End Sub
        Public Event PropertyChanged As Global.System.ComponentModel.PropertyChangedEventHandler Implements System.ComponentModel.INotifyPropertyChanged.PropertyChanged
        Protected Overridable Sub OnPropertyChanged(ByVal [property] As String)
            If (Me.PropertyChangedEvent IsNot Nothing) Then
                RaiseEvent PropertyChanged(Me, New Global.System.ComponentModel.PropertyChangedEventArgs([property]))
            End If
        End Sub
    End Class
End Namespace
